nautobot_golden_config/api/serializers.py

```diff
@@ -68,6 +69,8 @@ class GoldenConfigSettingSerializer(TaggedObjectSerializer, CustomFieldModelSeri
     url = serializers.HyperlinkedIdentityField(
         view_name="plugins-api:nautobot_golden_config-api:goldenconfigsetting-detail"
     )
+    scope = serializers.JSONField(required=False)
+    dynamic_group = NestedDynamicGroupSerializer(required=False)
```

Comment:
Can we add a validate on this class to ensure on write's that one of these is applied? 

For an example of that method in usage, see here: 
https://github.com/nautobot/nautobot/blob/dd660f7dbb2be1f50aa84a0e7c173f0f89bcd4be/nautobot/extras/api/nested_serializers.py#L169

Reply:
To me it is about mutually exclusive. Meaning, you cannot send both. I wasn't thinking about a version check.

Reply:
Going forward, all will have a scope that maps to a DynamicGroup. Starting in v1.4 DG with a filter of `{}` fail open instead of previously failing closed.